concordium-std/src/types.rs

```diff
@@ -672,6 +687,56 @@ pub type QueryAccountBalanceResult = Result<AccountBalance, QueryAccountBalanceE
 /// [`QueryContractBalanceError`] and result to [`Amount`].
 pub type QueryContractBalanceResult = Result<Amount, QueryContractBalanceError>;
 
+/// A wrapper around [`Result`] that fixes the error variant to
+/// [`QueryAccountPublicKeysError`] and result to [`AccountPublicKeys`].
+pub type QueryAccountPublicKeysResult = Result<AccountPublicKeys, QueryAccountPublicKeysError>;
+
+/// A wrapper around [`Result`] that fixes the error variant to
+/// [`CheckAccountSignatureError`] and result to [`bool`].
+pub type CheckAccountSignatureResult = Result<bool, CheckAccountSignatureError>;
+
+pub(crate) type KeyIndex = u8;
+
+#[derive(crate::Serialize, Debug, SchemaType)]
+pub(crate) enum PublicKey {
+    Ed25519(PublicKeyEd25519),
+}
+
+#[derive(crate::Serialize, Debug, SchemaType)]
+pub struct CredentialPublicKeys {
+    #[concordium(size_length = 1)]
+    pub(crate) keys:      crate::collections::BTreeMap<KeyIndex, PublicKey>,
+    pub(crate) threshold: SignatureThreshold,
+}
+
+#[derive(crate::Serialize, Debug, SchemaType)]
+pub struct AccountPublicKeys {
+    #[concordium(size_length = 1)]
+    pub(crate) keys:      crate::collections::BTreeMap<CredentialIndex, CredentialPublicKeys>,
+    pub(crate) threshold: AccountThreshold,
+}
+
+pub(crate) type CredentialIndex = u8;
+
+#[derive(crate::Serialize, Debug, SchemaType)]
+pub enum Signature {
```

Comment:
Consider marking it with [`#[non_exhaustive]`](https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute)

Reply:
It's a bit hard to know if that is valuable due to the way smart contracts are, new versions need to be deployed to be used.

But I added it.,